Index: app/src/main/java/com/example/susan/scanner/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/susan/scanner/MainActivity.java	(revision df26247f59ca35407364cd9bd409cb95483a6734)
+++ app/src/main/java/com/example/susan/scanner/MainActivity.java	(revision )
@@ -1,59 +1,79 @@
 package com.example.susan.scanner;
 
+import android.app.Activity;
 import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
 import android.widget.TextView;
+import android.widget.Toast;
 
+import com.google.zxing.integration.android.IntentIntegrator;
+import com.google.zxing.integration.android.IntentResult;
 
+
 public class MainActivity extends AppCompatActivity {
 
-    private static final int EDIT=1;
+    private TextView scanResult;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        TextView scanResult = (TextView)findViewById(R.id.textView3);
+        scanResult = (TextView)findViewById(R.id.textView3);
         scanResult.setText("");
 
         Button startBtn = (Button)findViewById(R.id.scanner);
+        final Activity activity = this;
         startBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Intent intent = new Intent();
-                intent.setClass(MainActivity.this, ScanActivity.class);
-                startActivityForResult(intent, EDIT);
+                IntentIntegrator integrator = new IntentIntegrator(activity);
+                integrator.setDesiredBarcodeFormats(IntentIntegrator.ONE_D_CODE_TYPES);
+                integrator.setPrompt("Scan");
+                integrator.setCameraId(0);
+                integrator.setBeepEnabled(true);
+                integrator.setBarcodeImageEnabled(false);
+                integrator.initiateScan();
             }
         });
     }
 
     @Override
-    public void onStart()
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        IntentResult result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data);
+        if(result != null)
-    {
+        {
-        super.onStart();
+            if(result.getContents() == null)
+            {
+                Toast.makeText(this, "You cancelled the scanning.", Toast.LENGTH_LONG).show();
+            }
+            else
+            {
+                String content = result.getContents();
+                Toast.makeText(this, content, Toast.LENGTH_LONG).show();
+                scanResult.append(content + "\n");
+            }
+        }
+        else
+        {
+            super.onActivityResult(requestCode, resultCode, data);
+        }
 
-
-
-
     }
 
     @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-
-        switch(requestCode){
-            case EDIT:
-                if(resultCode == RESULT_OK) {
-                    TextView scanResult = (TextView)findViewById(R.id.textView3);
-                    Bundle scanBundle = data.getExtras();
-                    String setText = scanBundle.getString("result") + "\n";
-                    scanResult.append(setText);
+    protected void onSaveInstanceState(Bundle outState)
+    {
+        outState.putString("textView", scanResult.getText().toString());
-                }
+    }
-        }
+
+    @Override
+    protected  void onRestoreInstanceState(Bundle savedInstanceState)
+    {
+        scanResult.setText(savedInstanceState.getString("textView"));
     }
 
 }
Index: app/src/main/java/com/example/susan/scanner/ScanActivity.java
===================================================================
--- app/src/main/java/com/example/susan/scanner/ScanActivity.java	(revision df26247f59ca35407364cd9bd409cb95483a6734)
+++ app/src/main/java/com/example/susan/scanner/ScanActivity.java	(revision df26247f59ca35407364cd9bd409cb95483a6734)
@@ -1,152 +0,0 @@
-package com.example.susan.scanner;
-
-import android.content.DialogInterface;
-import android.content.Intent;
-import android.content.pm.PackageManager;
-import android.os.Build;
-import android.support.v4.app.ActivityCompat;
-import android.support.v4.content.ContextCompat;
-import android.support.v7.app.AlertDialog;
-import android.support.v7.app.AppCompatActivity;
-import android.os.Bundle;
-import android.widget.Toast;
-
-import com.google.zxing.Result;
-import me.dm7.barcodescanner.zxing.ZXingScannerView;
-
-import static android.Manifest.permission.CAMERA;
-
-public class ScanActivity extends AppCompatActivity implements ZXingScannerView.ResultHandler {
-
-    private  static  final  int REQUEST_CAMERA = 1;
-    private ZXingScannerView scannerView;
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        scannerView = new ZXingScannerView(this);
-        setContentView(scannerView);
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
-        {
-            if(checkPermission())
-            {
-                Toast.makeText(ScanActivity.this, "Permission is granted!", Toast.LENGTH_LONG).show();
-            }
-            else
-            {
-                requestPermission();
-            }
-        }
-    }
-
-    private boolean checkPermission()
-    {
-        return (ContextCompat.checkSelfPermission(ScanActivity.this, CAMERA) == PackageManager.PERMISSION_GRANTED);
-    }
-
-    private void requestPermission()
-    {
-        ActivityCompat.requestPermissions(this, new String[] {CAMERA}, REQUEST_CAMERA);
-    }
-
-    public void onRequestPermissionResult(int requestCode, String permission[], int grantResult[])
-    {
-        switch(requestCode) {
-            case REQUEST_CAMERA:
-                if(grantResult.length > 0)
-                {
-                    boolean cameraAccepted = grantResult[0] == PackageManager.PERMISSION_GRANTED;
-                    if(cameraAccepted)
-                    {
-                        Toast.makeText(ScanActivity.this, "Permission is granted!", Toast.LENGTH_LONG).show();
-                    }
-                    else {
-                        Toast.makeText(ScanActivity.this, "Permission is Denied!", Toast.LENGTH_LONG).show();
-                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                            if (shouldShowRequestPermissionRationale(CAMERA)) {
-                                displayAlertMessage("You need to allow access for both permissions",
-                                        new DialogInterface.OnClickListener() {
-                                            @Override
-                                            public void onClick(DialogInterface dialogInterface, int i) {
-                                                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
-                                                    requestPermissions(new String[]{CAMERA}, REQUEST_CAMERA);
-                                            }
-                                        });
-                                return;
-                            }
-                        }
-                    }
-                }
-                break;
-        }
-    }
-
-    @Override
-    public void onResume()
-    {
-        super.onResume();
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
-        {
-            if(checkPermission())
-            {
-                if(scannerView == null)
-                {
-                    scannerView.resumeCameraPreview(this);
-                    setContentView(scannerView);
-                }
-                scannerView.setResultHandler(this);
-                scannerView.startCamera();
-            }
-            else
-            {
-                requestPermission();
-            }
-        }
-    }
-
-    @Override
-    public void onDestroy()
-    {
-        super.onDestroy();
-        scannerView.stopCamera();
-    }
-
-    public void displayAlertMessage(String message, DialogInterface.OnClickListener listener)
-    {
-        new AlertDialog.Builder(ScanActivity.this)
-                .setMessage(message)
-                .setPositiveButton("OK", listener)
-                .setNegativeButton("Cancel", null)
-                .create()
-                .show();
-    }
-
-    @Override
-    public void handleResult(Result result)
-    {
-        final String scanResult = result.getText();
-        AlertDialog.Builder builder = new AlertDialog.Builder(this);
-        builder.setTitle("Scan Result");
-        builder.setPositiveButton("Add", new DialogInterface.OnClickListener() {
-            @Override
-            public void onClick(DialogInterface dialog, int which) {
-                Intent intent = new Intent();
-                Bundle bundle = new Bundle();
-                bundle.putString("result", scanResult);
-                intent.putExtras(bundle);
-                setResult(RESULT_OK, intent);
-                finish();
-            }
-        });
-        builder.setNegativeButton("Discard", new DialogInterface.OnClickListener() {
-            @Override
-            public void onClick(DialogInterface dialog, int which) {
-                setResult(RESULT_CANCELED);
-                finish();
-            }
-        });
-        builder.setMessage(scanResult);
-        AlertDialog alert = builder.create();
-        alert.show();
-    }
-}
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/modules.xml	(revision df26247f59ca35407364cd9bd409cb95483a6734)
+++ .idea/modules.xml	(revision )
@@ -2,7 +2,9 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
+      <module fileurl="file://$PROJECT_DIR$/BarcodeScanner.iml" filepath="$PROJECT_DIR$/BarcodeScanner.iml" />
       <module fileurl="file://$PROJECT_DIR$/Scanner.iml" filepath="$PROJECT_DIR$/Scanner.iml" />
+      <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
     </modules>
   </component>
\ No newline at end of file
Index: app/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/AndroidManifest.xml	(revision df26247f59ca35407364cd9bd409cb95483a6734)
+++ app/src/main/AndroidManifest.xml	(revision )
@@ -21,7 +21,6 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
-        <activity android:name=".ScanActivity"></activity>
     </application>
 
 </manifest>
\ No newline at end of file
Index: app/src/main/res/layout/activity_scan.xml
===================================================================
--- app/src/main/res/layout/activity_scan.xml	(revision df26247f59ca35407364cd9bd409cb95483a6734)
+++ app/src/main/res/layout/activity_scan.xml	(revision df26247f59ca35407364cd9bd409cb95483a6734)
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context="com.example.susan.scanner.ScanActivity">
-
-</android.support.constraint.ConstraintLayout>
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(revision df26247f59ca35407364cd9bd409cb95483a6734)
+++ app/build.gradle	(revision )
@@ -24,8 +24,9 @@
     androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
-    compile 'me.dm7.barcodescanner:zxing:1.9'
     compile 'com.android.support:appcompat-v7:25.3.1'
+    compile 'com.google.zxing:core:3.3.0'
+    compile 'com.journeyapps:zxing-android-embedded:3.2.0@aar'
     compile 'com.android.support.constraint:constraint-layout:1.0.2'
     testCompile 'junit:junit:4.12'
 }
